charts/linkerd2/templates/controller.yaml

```diff
@@ -81,8 +81,8 @@ spec:
         - -cluster-domain={{.Values.global.clusterDomain}}
         {{- if .Values.global.prometheusUrl }}
         - -prometheus-url={{.Values.global.prometheusUrl}}
-        {{- else if .Values.prometheus.enabled }}
-        - -prometheus-url=http://linkerd-prometheus.{{.Values.global.namespace}}.svc.{{.Values.global.clusterDomain}}:9090
+        {{- else }}
+        - -prometheus-url=http://linkerd-prometheus.{{.Values.linkerdVizNamespace}}.svc.{{.Values.global.clusterDomain}}:9090
```

Comment:
We won't need this once the controller has moved into the linkerd-viz extension since it will be able to use the `prometheusAddr` value in that extension directly.  For now, I think we should just hardcode the namespace here.